src/reconcile/rec_visibility.c

```diff
@@ -582,10 +613,10 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
          * for the data store. Subsequent reconciliations should know that they can select this
          * update regardless of visibility.
          */
-        if (upd_select->upd != NULL)
-            F_SET(upd_select->upd, WT_UPDATE_DS);
-        if (tombstone != NULL)
-            F_SET(tombstone, WT_UPDATE_DS);
+        if (upd_select->upd != NULL && !F_ISSET(upd_select->upd, WT_UPDATE_DS))
+            __rec_add_datastore_update(session, r, upd_select->upd);
```

Comment:
error handling if this returns an error, in case "__wt_calloc_one" failed.

Reply:
01c2d22